tasks/configure.yml

```diff
@@ -152,10 +158,38 @@
     owner: "{{ postgresql_service_user }}"
     group: "{{ postgresql_service_group }}"
     mode: u=rwX,g=rwXs,o=rx
-  notify: restart postgresql with service
+  notify: restart postgresql
+
+- name: PostgreSQL | Enable service
+  service:
+    name: "{{ postgresql_service_name }}"
+    enabled: yes
+  when: postgresql_service_enabled
+
+- name: PostgreSQL | Ensure PostgreSQL is running
+  service:
+    name: "{{ postgresql_service_name }}"
+    state: started
+  register: postgresql_cold_started
 
 - name: PostgreSQL | Reload all conf files
   service:
     name: "{{ postgresql_service_name }}"
     state: reloaded
-  when: postgresql_configuration_pt1.changed or postgresql_configuration_pt2.changed or postgresql_configuration_pt3.changed or postgresql_systemd_custom_conf.changed
+  notify: reload postgresql
+  when: not postgresql_cold_started.changed and
+        ( postgresql_configuration_pt1.changed or
+          postgresql_configuration_pt2.changed or
+          postgresql_configuration_pt3.changed or
+          postgresql_systemd_custom_conf.changed)
+
+- name: PostgreSQL | Check for parameter changes that require a restart | >= v9.5
+  become: yes
+  become_user: "{{ postgresql_service_user }}"
+  shell: "{{ postgresql_bin_directory }}/psql postgres --username {{ postgresql_admin_user }} --tuples-only --command='SELECT count(*) COUNT FROM pg_settings WHERE pending_restart IS true;'"
+  register: postgresql_check_for_param_changes
+  notify: reload postgresql
```

Comment:
I'm notifying `reload`, not `restart`... so that that handler can decide what to do depending on the version of PostgreSQL, etc.